mssqlcli/mssql_cli.py

```diff
@@ -366,6 +368,20 @@ def execute_query(self, text):
             output = self._execute_interactive_command(text)
         else:
             # non-interactive mode
+            if self.input_file:
+                if text:
+                    # return error if both query text and an input file are specified
+                    click.secho("Invalid arguments: either -Q or -i may be specified.",
```

Comment:
Minor: a common pattern is to validate fields at the start (e.g. -Q vs -i) and then have streamlined logic. This can make it cleaner as additional requirements come up later. Not required, just something to think about

Reply:
Thanks @kevcunnane, agreed. I'll look into moving functionality to constructor.